stellargraph/core/graph.py

```diff
@@ -575,31 +589,33 @@ def _transform_edges(
             if weights is not None:
                 weights = weights[correct_type]
 
-        # FIXME(#718): it would be better to return these as ndarrays, instead of (zipped) lists
         if weights is not None:
+            return other_node_id, weights
+
+        return other_node_id
+
+    def _to_neighbors(self, neigh_arrs, include_edge_weight):
+        if include_edge_weight:
             return [
-                NeighbourWithWeight(node, weight)
-                for node, weight in zip(other_node_id, weights)
+                NeighbourWithWeight(neigh, weight) for neigh, weight in zip(*neigh_arrs)
             ]
+        return list(neigh_arrs)
 
-        return list(other_node_id)
-
-    def neighbors(
+    def neighbor_arrays(
         self, node: Any, include_edge_weight=False, edge_types=None
-    ) -> Iterable[Any]:
+    ) -> Iterable[np.ndarray]:
         """
         Obtains the collection of neighbouring nodes connected
         to the given node.
```

Comment:
I think the docstrings for these new methods should be slightly different from the existing methods (similar to how you've done `edge_arrays`), otherwise I'm not sure if it's obvious from just the names of the methods how they're different? Maybe something like
```
Obtains the collection of neighbouring nodes connected to the given node as a tuple of arrays (sources, targets, types, weights). ``types`` and ``weights`` will be None unless specified via the optional parameters.

...

Returns:
    (sources, targets, types, weights)
```
And similarly for `in_node_arrays`, `out_node_arrays`